Demo/src/main/java/com/braintreepayments/demo/GooglePayFragment.java

```diff
@@ -99,8 +105,14 @@ public void launchGooglePay(View v) {
                 .addAllowedCountryCodes(Settings.getGooglePayAllowedCountriesForShipping(activity))
                 .build());
 
-        googlePayClient.createPaymentAuthRequest(googlePayRequest,
-                (paymentAuthRequest, error) -> googlePayLauncher.launch(paymentAuthRequest));
+        googlePayClient.createPaymentAuthRequest(googlePayRequest, (paymentAuthRequest) -> {
+            if (paymentAuthRequest instanceof GooglePayPaymentAuthRequest.ReadyToLaunch) {
+                googlePayLauncher.launch(
+                        ((GooglePayPaymentAuthRequest.ReadyToLaunch) paymentAuthRequest).getRequestParams());
+            } else if (paymentAuthRequest instanceof GooglePayPaymentAuthRequest.Failure) {
```

Comment:
are there any other possible values for `GooglePayPaymentAuthRequest`? Curious if we need a fallback for scenarios we don't catch.

Reply:
These are the only options - the benefit of defining `GooglePayPaymentAuthRequest` as a sealed class is that it is exhaustive, and when we migrate this file to Kotlin, we can use the `when` switch statement without a default/fallback scenario.

Reply:
> Curious if we need a fallback for scenarios we don't catch.

This is one thing that keeps me up at night lol. Adding a new case to an existing enum can totally break a merchant app. The tradeoff is reduced compiler enforcement in exchange for not breaking a merchant integration.

I do think it's worth calling out in our docs to encourage merchants to catch "UNKNOWN" results so that their integrations are forward compatible.